pkg/operators/spoke/controller.go

```diff
@@ -228,86 +229,84 @@ func (n *nucleusSpokeController) sync(ctx context.Context, controllerContext fac
+			Type: spokeCoreApplied, Status: metav1.ConditionFalse, Reason: "SpokeCoreApplyFailed",
 			Message: fmt.Sprintf("Failed to deploy registration deployment with error %v", err),
-		})
-		helpers.UpdateNucleusSpokeStatus(
-			ctx, n.nucleusClient, spokeCoreName, helpers.UpdateNucleusSpokeConditionFn(spokeCore.Status.Conditions...))
+		}))
 		return err
 	}
+	// TODO store this in the status of the spokecore itself
 	n.registrationGeneration = generation
 
+	// Deploy work agent
+	generation, err = n.applyDeployment(
+		config, "manifests/spoke/spoke-work-deployment.yaml", n.workGeneration, controllerContext)
+	if err != nil {
+		helpers.UpdateNucleusSpokeStatus(ctx, n.nucleusClient, spokeCoreName, helpers.UpdateNucleusSpokeConditionFn(nucleusapiv1.StatusCondition{
+			Type: spokeCoreApplied, Status: metav1.ConditionFalse, Reason: "SpokeCoreApplyFailed",
+			Message: fmt.Sprintf("Failed to deploy work deployment with error %v", err),
+		}))
+		return err
+	}
+	// TODO store this in the status of the spokecore itself
+	n.workGeneration = generation
+
+	// if we get here, we have successfully applied everything and should indicate that
+	helpers.UpdateNucleusSpokeStatus(ctx, n.nucleusClient, spokeCoreName, helpers.UpdateNucleusSpokeConditionFn(nucleusapiv1.StatusCondition{
+		Type: spokeCoreApplied, Status: metav1.ConditionTrue, Reason: "SpokeCoreApplied",
+		Message: "Spoke Core Component Applied",
+	}))
+
+	// now that we have applied all of our logic, we can check to see if the data we expect to have present as indications of
+	// proper functioning of registration controller is working
+	// TODO this should be moved into a separate loop since it is independent of the application of the eventually consistent
+	//  resources above
+
 	// If cluster name is empty, read cluster name from hub config secret
 	if config.ClusterName == "" {
 		clusterName := hubSecret.Data["cluster-name"]
 		if clusterName == nil {
-			helpers.SetNucleusCondition(&spokeCore.Status.Conditions, nucleusapiv1.StatusCondition{
-				Type:    spokeCoreApplied,
-				Status:  metav1.ConditionFalse,
-				Reason:  "SpokeCoreApplyFailed",
-				Message: fmt.Sprintf("Failed to get cluster name"),
-			})
-			helpers.UpdateNucleusSpokeStatus(
-				ctx, n.nucleusClient, spokeCoreName, helpers.UpdateNucleusSpokeConditionFn(spokeCore.Status.Conditions...))
+			helpers.UpdateNucleusSpokeStatus(ctx, n.nucleusClient, spokeCoreName, helpers.UpdateNucleusSpokeConditionFn(nucleusapiv1.StatusCondition{
+				Type: spokeRegistrationDegraded, Status: metav1.ConditionTrue, Reason: "ClusterNameMissing",
+				Message: fmt.Sprintf("Failed to get cluster name from `kubectl get secret -n %q %q -ojsonpath='{.data.cluster-name}`.  This is set by the spoke registration deployment.", hubSecret.Namespace, hubSecret.Name),
+			}))
 			return fmt.Errorf("Failed to get cluster name")
 		}
 		config.ClusterName = string(clusterName)
```

Comment:
I think we should get cluster name from secret before deploying work agent. If cluster name is not specified in spokecore CR, we will not be able to update the deployment.